### PR TITLE
Chameleon Contractor Baton

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Allows the contractor baton to disguise as iron rods, until the baton is used, at which point it automatically undisguises.

## How This Contributes To The Skyrat Roleplay Experience
It allows the contractor to get a jump on targets without being exposed by the baton's sprite (the sound is still audible for those with sharp ears)

## Changelog

:cl:
balance: Contractor baton now has chameleon functionality
/:cl:

### CODE REVIEW

**Comment by tf-4** [comment]
👎 already the best baton in the game (well abductor baton too but you get my point)

File: modular_skyrat/modules/contractor/code/datums/baton_disguise.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	baton.lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	baton.righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	baton.inhand_icon_state = "rods"
+	baton.name = "iron rod"
+	baton.icon = 'icons/obj/stack_objects.dmi'
+	baton.icon_state = "rods"
+	disguised = TRUE
+	owner.regenerate_icons()
+	baton.update_appearance()
+	baton.balloon_alert(owner, "baton disguised")
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/contractor/code/datums/baton_disguise.dm]: might wanna make it update desc too? idk

**Comment by Zonespace27** [comment]
> 👎 already the best baton in the game (well abductor baton too but you get my point)

Currently the contractor's kit is fully able to disguise off the bat (modsuit chameleon module, use a data disk to move the contract uplink from the contractor tablet to a PDA, etc.) with the exception of the baton (unless you buy the baton holster, which is its own investment and comes with its own drawbacks), though I could probably throw in a doubled cooldown time for use of the baton when in chameleon mode

File: modular_skyrat/modules/contractor/code/datums/baton_disguise.dm (hunk 1 above)
Comment (Zonespace27) [reply modular_skyrat/modules/contractor/code/datums/baton_disguise.dm]: Gonna take a look at refactoring this tomorrow, I wrote this pretty quickly and the code quality shows

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by tf-4** [comment]
Still not a fan at all, it doesn't need any buffs. The baton retracts, that's how you can disguise it. This is too much. The first hit is the most important.

**Comment by Zonespace27** [comment]
Eh

**Pull request closed without merging**